src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1863,6 +1860,11 @@ void ObjectMonitor::wait(jlong millis, bool interruptible, TRAPS) {
   assert(!has_successor(current), "invariant");
   assert_mark_word_consistency();
 
+  if (ce != nullptr && ce->is_virtual_thread()) {
+    assert(result != freeze_ok, "sanity check");
```

Comment:
This assertion is so far from where `result` is set that I don't think it serves any purpose. We would have returned at line 1693 already. It was already a long way before you moved this.

Reply:
It would be bug to call JavaThread::post_vthread_pinned_event with freeze_ok so the asserts at both call sites were to catch that. Since this function has an assert too then they are redundant at the usages so can be removed.